tests/protocol/test_goal_state.py

```diff
@@ -57,4 +64,40 @@ def test_update_goal_state_should_save_goal_state(self):
         extensions = vm_settings["extensionGoalStates"]
         self.assertEqual(4, len(extensions), "Expected 4 extensions in the test vmSettings")
         for e in extensions:
-            self.assertEqual(e["settings"][0]["protectedSettings"], "*** REDACTED ***", "The protected settings for {0} were not redacted".format(e["name"]))
\ No newline at end of file
+            self.assertEqual(e["settings"][0]["protectedSettings"], "*** REDACTED ***", "The protected settings for {0} were not redacted".format(e["name"]))
+
+
+    def test_update_extensions_goal_state_should_should_retry_on_resource_gone_error(self, _):
+        """
+        Requests to the hostgaplugin incude the Container ID and the RoleConfigName as headers; when the hostgaplugin returns GONE (HTTP status 410) the agent
+        needs to get a new goal state and retry the request with updated values for the Container ID and RoleConfigName headers.
```

Comment:
Maybe add this comment on the main code above too? Just to make the intension clearer for the uninitialized

Reply:
i'll probably refactor the retry in the hostgaplugin class itself and i'll document it there; will also see if I can remove some of the duplicates of this retry logic